idaes/ui/fsvis/static/js/paper.js

```diff
@@ -63,6 +63,34 @@ export class Paper {
         // context menu
         self._paper.on("element:contextmenu", function(cellView, evt) {
             cellView.model.rotate(90)
+            // This is needed to keep the text labels for the unit models in the correct orientation
+            // x and y were specifically picked to keep the label in the same place 
+            // in relation to the unit model (bottom middle)
+            let x = 0
+            let y = 0
+            const angle = cellView.model.angle()
+            switch (angle) {
+                case 0:
```

Comment:
where do these "magic" numbers come from? Are all icons the same width and height? (If so, shouldn't they at least have different widths and heights..?)

Reply:
The numbers come from me manually changing the numbers bit by bit until the labels stay in the same place.

Yes, all of the icons are the same width and height. That's something that we knew needed to happen but it was put off as low priority.